### ISSUE TITLE
no-cycle crashes with stack overflow on self-importing files

### ISSUE THREAD

**Issue by silverwind** [nebrius/eslint-plugin-fast-import#45]
The `no-cycle` rule crashes with `RangeError: Maximum call stack size exceeded` when a file imports itself. The `traverse` function in `computeAnalyzedInfo.js` recurses infinitely because it doesn't detect a file importing itself.

## Reproduction

```js
// eslint.config.js
import { recommended } from "eslint-plugin-fast-import";
import { dirname } from "node:path";
import { fileURLToPath } from "node:url";

export default [
  recommended({
    rootDir: dirname(fileURLToPath(import.meta.url)),
  }),
];
```

```ts
// self-import.ts
import { foo } from "./self-import.ts";
export const foo = 1;
```

```
$ npx eslint self-import.ts
```

## Expected behavior

The `no-cycle` rule reports a cycle error on the self-import.

## Actual behavior

```
Oops! Something went wrong! :(

ESLint: 9.39.2

RangeError: Error while loading rule 'fast-import/no-cycle': Maximum call stack size exceeded
Occurred while linting self-import.ts
    at file:///node_modules/eslint-plugin-fast-import/dist/module/computeAnalyzedInfo.js:196:76
    at Array.find (<anonymous>)
    at traverse (file:///node_modules/eslint-plugin-fast-import/dist/module/computeAnalyzedInfo.js:196:71)
    at traverse (file:///node_modules/eslint-plugin-fast-import/dist/module/computeAnalyzedInfo.js:219:25)
    at traverse (file:///node_modules/eslint-plugin-fast-import/dist/module/computeAnalyzedInfo.js:219:25)
    ...
```

## Environment

- eslint-plugin-fast-import: 1.10.0
- eslint: 9.39.2
- node: 24.13.1
- os: macOS (darwin arm64)

**Comment by nebrius** [comment]
Nice find, I honestly never thought to test this case. Bug is fixed in version 1.11.0
